Router/SpecialRouter.php

```diff
@@ -29,26 +30,61 @@ class SpecialRouter extends Base
     ];
 
     /**
-     * Translate special characters in the path to names.
+     * Match a route to this router.
+     *
+     * If there is a match, replace the path with the new path.
+     * Ex. /my-module/foo-bar/baz_qux -> /my-module/foodashbar/bazunderscorequx
      *
      * @param RequestInterface $request
-     * @return array
+     * @return ActionInterface|null
+     */
+    public function match(
+        RequestInterface $request,
+    ): ActionInterface|null {
+        $identifier = trim($request->getPathInfo(), '/');
+        $pathParts = explode('/', $identifier);
+        $moduleName = array_shift($pathParts);
+        $pathInfo = implode('/', $pathParts);
+
+        if ($this->isMatch($pathInfo)) {
+            $newPathInfo = $this->replacePath($pathInfo);
+            $newPathInfo = "/$moduleName/$newPathInfo";
+            $request->setPathInfo($newPathInfo);
+        }
+
+        // Now that path has been updated, return null to allow the next router to match.
```

Comment:
> Now that path has been updated

No, it only updates if it matches: `if ($this->isMatch($pathInfo)) {}`

Reply:
@TuVanDev updated 👍